src/components/legend/draw.js

```diff
@@ -154,24 +154,39 @@ function drawOne(gd, opts) {
         function() {
             var gs = fullLayout._size;
             var bw = legendObj.borderwidth;
+            var isPaperX = legendObj.xref === 'paper';
+            var isPaperY = legendObj.yref === 'paper';
 
             if(!inHover) {
-                var expMargin = expandMargin(gd, legendId);
+                var lx, ly;
+
+                if(isPaperX) {
+                    lx = gs.l + gs.w * legendObj.x - FROM_TL[getXanchor(legendObj)] * legendObj._width;
+                } else {
+                    legendObj.x = Lib.constrain(legendObj.x, 0, 1); // TODO: Move this to defaults setting?
```

Comment:
Oh and I guess this `Lib.constrain` can be removed now (as well as the TODO below on line 173)